plugin/pkg/scheduler/schedulercache/cache_test.go

```diff
@@ -0,0 +1,484 @@
+			t.Fatalf("assumePod failed: %v", err)
+		}
+		cache.cleanupAssumedPods(now.Add(2 * ttl))
+		// It should be expired and removed.
+		n := cache.nodes[nodeName]
+		if n != nil {
+			t.Errorf("#%d: expecting nil node info, but get=%v", i, n)
+		}
+		if err := cache.AddPod(tt.pod); err != nil {
+			t.Fatalf("AddPod failed: %v", err)
+		}
+		// check after expiration. confirmed pods shouldn't be expired.
+		n = cache.nodes[nodeName]
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestUpdatePod tests that a pod will be updated if added before.
+func TestUpdatePod(t *testing.T) {
+	nodeName := "node"
+	ttl := 10 * time.Second
+	testPods := []*api.Pod{
+		makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}}),
+		makeBasePod(nodeName, "test", "200m", "1Ki", []api.ContainerPort{{HostPort: 8080}}),
+	}
+	tests := []struct {
+		// 0: pod to assume and add
+		// 1, 2: pod to update
+		pods []*api.Pod
+
+		wNodeInfo  *NodeInfo
+		wNodeInfo2 *NodeInfo
+	}{{ // Pod is assumed and added. Then it would be updated twice.
+		pods: []*api.Pod{testPods[0], testPods[1], testPods[0]},
+		wNodeInfo: &NodeInfo{
+			requestedResource: &Resource{
+				MilliCPU: 200,
+				Memory:   1024,
+			},
+			nonzeroRequest: &Resource{
+				MilliCPU: 200,
+				Memory:   1024,
+			},
+			pods: []*api.Pod{testPods[1]},
+		},
+		wNodeInfo2: &NodeInfo{
+			requestedResource: &Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			nonzeroRequest: &Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			pods: []*api.Pod{testPods[0]},
+		},
+	}}
+
+	now := time.Now()
+	for i, tt := range tests {
+		cache := newSchedulerCache(ttl, time.Second, nil)
+		if err := cache.assumePodIfBindSucceed(tt.pods[0], alwaysTrue, now); err != nil {
+			t.Fatalf("assumePod failed: %v", err)
+		}
+		if err := cache.AddPod(tt.pods[0]); err != nil {
+			t.Fatalf("AddPod failed: %v", err)
+		}
+		if err := cache.UpdatePod(tt.pods[0], tt.pods[1]); err != nil {
+			t.Fatalf("UpdatePod failed: %v", err)
+		}
+		// check after expiration. confirmed pods shouldn't be expired.
+		n := cache.nodes[nodeName]
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+
+		// unlike Add, Update can happen multiple times
+		if err := cache.UpdatePod(tt.pods[1], tt.pods[2]); err != nil {
+			t.Fatalf("UpdatePod failed: %v", err)
+		}
+		// check after expiration. confirmed pods shouldn't be expired.
+		n = cache.nodes[nodeName]
+		if !reflect.DeepEqual(n, tt.wNodeInfo2) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestRemovePod tests after added pod is removed, its information should also be subtracted.
+func TestRemovePod(t *testing.T) {
+	nodeName := "node"
+	basePod := makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}})
+	tests := []struct {
```

Comment:
Similar to above - you are testing a specific single scenario here - it doesn't make sense to have an array of structs here (this is misleading)